### PR TITLE
Use custom field descriptions if available

### PR DESCRIPTION
**Pull request by nacnudus**
### Context

https://github.com/openregister/deployment/issues/514

### Changes proposed in this pull request

When creating RSF with `scripts/rsfcreator.py` (called by several other scripts, in particular `scripts/load-register-tsv.sh`) look for a custom field definition in `registry-data/data/{phase}/{register}/field/{field}.yaml` and use it to override the definition in `registry-data/data/{phase}/field/{field}.yaml`.

A test is provided.

### Guidance to review

Check that it implements the first "alternative" discussed in the #514

### CODE REVIEW

**Comment by nacnudus** [comment]
Now it overrides individual properties of fields, rather than the whole thing.

Not sure if `.__contains__()` is pythonic, what do you think @arnau?

File: scripts/rsfcreator.py [hunk 1]
```diff
@@ -77,14 +75,20 @@ def rsf_for_line(line_dict, key_field, entry_type, key_prefix='', key=None):
             entry_type, key_prefix, key, timestamp, item_hash)
     return (item_line, entry_line)
 
+def merge_level_2(dct, other):
+    for k, v in other.items():
+        if (other.__contains__(k)):
```
Comment (arnau) [inline scripts/rsfcreator.py]: The use of underscored methods should be avoided as much as possible. A common way to check the same would be:

```python
if k in other:
  dct[k] ...
```

File: scripts/rsfcreator.py [hunk 2]
```diff
@@ -77,14 +75,20 @@ def rsf_for_line(line_dict, key_field, entry_type, key_prefix='', key=None):
...
+
 def generate_rsf(args):
     if args.register_data_root:
         register_def = read_register_from_local(args.phase, args.register_name, args.register_data_root)
         field_names = register_def['fields']
         fields_by_name = {fn: read_field_from_local(args.phase, fn, args.register_data_root) for fn in field_names}
         custom_fields_by_name = {fn: read_custom_field_from_local(args.phase, args.register_name, fn, args.register_data_root) for fn in field_names}
-        custom_fields_by_name = {k: v for k, v in custom_fields_by_name.items() if v}
-        fields_by_name = {**fields_by_name, **custom_fields_by_name}
+        custom_fields_by_name = remove_blanks(custom_fields_by_name)
```
Comment (arnau) [inline scripts/rsfcreator.py]: On second thought, perhaps the two expressions that generate `custom_fields_by_name` could be extracted as a function so the logic is isolated (and can be tested as well), something on the lines of:

```python
def custom_fields(phase, register_name, register_data_root, field_names):
    fields = {}
    for name in field_names:
        field = read_custom_field_from_local(phase, register_name, name, register_data_root)
        if field is not None:
            fields[name] = field

    return fields

def generate_rsf(args):
  ...
  custom_fields_by_name = custom_fields(args.phase, args.register_name,  args.register_data_root, field_names)
```

**Comment by nacnudus** [comment]
There's now `read_global_field_from_local()`, `read_custom_field_from_local()`, and a wrapper `read_field_from_local()` that merges global with custom when a custom one exists.

The deep merge function is no longer needed.

**Pull request merged**